File: ProjectStart/ProjectStart.WebApp/Controllers/Admin/UserController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using ProjectStart.Entity;
using ProjectStart.Repository;
using ProjectStart.ViewModel.AccountViewModels;
using AutoMapper;
using ProjectStart.Common.ViewModel;
using ProjectStart.Common;
using ProjectStart.Common.Extensions;

namespace ProjectStart.WebApp.Controllers.Admin
{
    [Area("Admin")]
    [Produces("application/json")]
    public class UserController : BaseController
    {
        public UserController(IUnitOfWork currentService, IMapper currentMapper) : base(currentService, currentMapper)
        { }
        public IActionResult Index()
        {
            return View("MainBody", null);
        }
        [HttpGet]
        public Response<FilterDataViewModel<UserViewModel>> GetUser(
            [FromQuery] FilterViewModel filterQuery
            )
        {
            var filterData = CurrentService.UserRepository.Entity.Select(s => CurrentMapper.Map<UserViewModel>(s)).OrderBy(o=>o.FirstName).AsQueryable().Paginate(filterQuery);

            return Success(filterData);
        }
    }
}

File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/UserController.ts
import { FilterDataViewModel } from 'shared/AppModels/FilterDataViewModel';
import { FilterQueryViewModel } from 'shared/AppModels/FilterQueryViewModel';
import { UserViewModel } from 'shared/AppModels/UserViewModel';
import { Api } from 'shared/Component/api';
import Response from 'shared/Component/response';


// State for post: api/User/
// post: api/User/
export const postIndexActionId = 'ProjectStart.WebApp.Controllers.Admin.UserController.Index.post';
export const postIndex = (localData?: (data: any) => void): Promise<any> => Api(postIndexActionId, 'post', `/api/User/`, null, true , localData);
// State for get: api/User/
// get: api/User/
export const getUserActionId = 'ProjectStart.WebApp.Controllers.Admin.UserController.GetUser.get';
export const getUser = (filterQuery: FilterQueryViewModel, localData?: (data: Response<FilterDataViewModel<UserViewModel>>) => void): Promise<Response<FilterDataViewModel<UserViewModel>>> => Api(getUserActionId, 'get', `/api/User/`, filterQuery, true , localData);

